web/src/rendered_markdown.ts

```diff
@@ -212,7 +212,7 @@ export const update_elements = ($content: JQuery): void => {
         if (user_group_id && $(this).find(".highlight").length === 0) {
             // Edit the mention to show the current name for the
             // user group, if its not in search.
-            set_name_in_mention_element(this, user_group.name);
+            set_name_in_mention_element(this, user_groups.get_display_group_name(user_group.name));
```

Comment:
Let's pull this out of this PR, since it's not required and likely needs to be reworked.

Reply:
Done.